core/services/fluxmonitorv2/flux_monitor.go

```diff
@@ -466,12 +466,16 @@ func formatTime(at time.Time) string {
 // SetOracleAddress sets the oracle address which matches the node's keys.
 // If none match, it uses the first available key
 func (fm *FluxMonitor) SetOracleAddress() error {
+
+	// fm on deprecation path, using dangling context
+	ctx := context.Background()
```

Comment:
We can still use the `fm.chStop` to trivially create a context that won't block shutdown.

Reply:
Just to make sure how I'm thinking about things is correct - why would it block shutdown if there's no logic in `SetOracleAddress()` that is blocking on a Done signal from the context?

Reply:
> why would it block shutdown if there's no logic in SetOracleAddress() that is blocking on a Done signal from the context?

The call to `EnabledKeysForChain` can block.